llama_index/core/base_retriever.py

```diff
@@ -144,7 +167,7 @@ def _handle_recursive_retrieval(
             node = n.node
             score = n.score or 1.0
             if isinstance(node, IndexNode):
-                obj = self.object_map.get(node.index_id, None)
+                obj = node.obj or self.object_map.get(node.index_id, None)
```

Comment:
help me understand, is `object_map` now only used for retrievers/query engines? 

If it's a Node it should now be serialized/deserialized directly on the `IndexNode` right? 

at a high-level once we make retrievers/query engine serializable i was thinking object_map would go away, and we'd replace with a proper docstore

Reply:
Yes if query engines/retrievers were serializable, this would go away.

Right now, unserializable index nodes have to be passed in under the `objects` kwarg -- from there, we can build a map of index id to object 

Then we can serialize and retrieve the index node without the object.

If an index node is retrieved, the object map is checked if we have its object